.changeset/funny-countries-hang.md

```diff
@@ -0,0 +1,5 @@
+---
+"@latticexyz/world": patch
+---
+
+Replaced the `systemFunctionSelector` argument in `registerRootFunctionSelector` with `systemFunctionSignature`, so it can be registered onchain.
```

Comment:
```suggestion
`registerRootFunctionSelector` now expects a `systemFunctionSignature` instead of a `systemFunctionSelector`. Internally, we compute the selector from the signature. This allows us to track system function signatures that are registered at the root so we can later generate ABIs for these systems.
```